File: Swift/helloworld.swift
import Foundation

print("Hola, Swift!")

var myString = "cadena de texto"
print("La longitud de la cadena es \(myString.count) caracteres.")
print(myString)

var myFloat: Float = 3.1416
print("El valor de myFloat es \(myFloat)")

let myConstatnt = "Constante"
print("El valor de la constante es \(myConstatnt)")

var myOptional: String? = "opcional"
myOptional = nil

if myFloat > 3.0 {
    print("myFloat es mayor que 3.0")
} else {
    print("myFloat no es mayor que 3.0")
} 

var myList = ["Leandro", "leandmartine", "Desarrollo"] // debe de ser una lista del mismo tipo de dato
print("Elementos de la lista \(myList)")
myList.append("iOS")
print(myList)

var mySet: Set = [1, 2, 3, 4, 1] // no es ordenado, no se puede repetir valores. 
print("Elementos del set \(mySet)")

var myMap = [
    "nombre": "Leandro",
    "usuario": "leandmartine",
    "curso": "Desarrollo iOS"
] // tiene que seguir el primer tipado, entidad string, valor string.
print("Elementos del diccionario \(myMap)")
print("El nombre es \(myMap["nombre"])")

if let usuario = myMap["usuarioS"] {
    print("El usuario es \(usuario)")
} else {
    print("El usuario no existe")
}
print(myMap["usuario"] ?? "El usuario no existe") // operador nil coalescing

for value in myList {
    print("Valor de la lista \(value)")
}

var index = 0
while index < myList.count {
    print("Valor de la lista en while \(myList[index])")
    index += 1
  
}

// Definición de función con -> se asigna el retorno de esta funcion
func myFunction() -> String{
  print("Esta es mi función")

  return "Función ejecutada"
}

var retorno = myFunction()

class MyClass {
    var myProperty: String? = "Propiedad de la clase"
    
    func myMethod() {
        print("Este es un método de la clase")
    }
}

var myClass = MyClass()
print(myClass.myProperty ?? "Propiedad nula")
myClass.myMethod()

File: Python/helloworld.py
print ("Hola, mundo")

'''
Este es un comentario multilinea
aaa
'''

my_string = "Hola, mundo"
my_string = "cambio valor variable"
print (my_string)

my_string = 0 # se cambia el tipo de dato
print (my_string)

my_int = 9
my_int = my_int + 1
print (my_int - 20)

my_float = 1.5
print (my_float * my_int)

my_bool = True
print (my_bool)

print(f"El valor de mi int es: {my_int}, y el valor de mi float es: {my_float}")

my_list = [my_string, my_bool, my_float, my_int]
print (my_list)
print (my_list[2])

my_dict = {"nombre": "Juan",
"edad": 30,
"ciudad": "Madrid"}

print (my_dict)
print (my_dict["nombre"])

#sets valores unicos // no se pueden repetir
#estructura desordenada
my_tuple = (my_string, my_bool, my_float, my_int)
print (my_tuple)

if my_int > 5 and my_float < 2:
    print ("Mi int es mayor que 5")
elif my_int == 5:
    print ("Mi int es igual que 5")
else:
    print ("Mi int es menor o igual que 5")

for my_item in my_list:
    print (my_item)

def my_function():
  print ("Esto es una funcion")

my_function()

def my_function_with_return():
    return "Esto es una funcion con return"

my_var = my_function_with_return()
print (my_var)

class MyFirstClass:
    my_name = "Clase 1"

    def __init__(self, my_name):
      self.my_name = my_name

    def print_name(self):
        print (self.my_name)

my_class= MyFirstClass("Clase Instanciada")
print (my_class.my_name)
MyFirstClass.my_name = "Clase Modificada"
print (my_class.my_name)
